### ISSUE TITLE
Initial blinds state upon homebridge restart

### ISSUE THREAD

**Issue by unscel** [AtomBaf/homebridge-mqtt-blinds-tasmota#1]
Hi AtomBaf, first of all I wanted to thank you for the great work. I was hovering around the web if I can fit a more simple way to control my custom automation solution via homekit for the balcony motorized shade. 

I was quite surprized the way how you applied the RF comm. via tasmotized sonoff 4ch as it is the same approach I followed due to rare 433.42mhz RF frequency of my shade. I simply wired up a cheap chineese RF remote buttons to 4ch and used pulsetime in tasmota to trigger the movements. Long story short, thanks to your plugin everything worked perfectly in homebridge/homekit out of the box, with no modification needed etiher in tasmotized 4ch or in the plugin code. So, managed to get an elegant solution in homekit instead of going back and forth around multiple momentary switches. 

And here is the question. Each time homebridge instance restarts, with your plugin, the state of the shade/blind remains open regardless of the last known state. Is it the same behavior you observe and if there is a way to avoid/reverse default initial open state or better to retain last known state via mqtt?  As most of the time I keep the shade closed that gives me false positives about the state of the shade.

Thank you again for the great job and have a great day.

**Comment by AtomBaf** [comment]
Hi unscel,

Yes I'm having the same issue, and don't bother much because at home my shutters are mostly opened. Since there is no position feedback, I simply ignored this issue, thinking that there was no solution.
But afaik you're right, MQTT could have the state recorded and I can probably read this state upon restart of the homebridge server.

Let me have a look, I'll hopefully have some time this weekend.

Many thanks for your feedback

**Comment by AtomBaf** [comment]
Hi again, 2 things have been added:
 - the plugin will now listen the telemetry events comming from tasmota. This will ensure the state to be in-sync wherever happen
 - during initialization, however, we should not wait for the telemetry event which can occur minutes after. So I trigger a MQTT command to force the shutter to publish its state

New configuration variable "mqttTeleTopic" available, useful to override the default which shoud work as-is for you. I just upgraded/restarted my homebridge and everything is okay.
Thanks for the tip
